### PR TITLE
Added separate executor list to handle CDS unauthenticated endpoints

### PR DESCRIPTION
**Pull request by aka4rKO**
## Added separate executor list to handle CDS unauthenticated endpoints

> $subject by not engaging the MTLS executor for the following unauthenticated elected resources:
"/banking/products", "/banking/products/{productId}", "/discovery/status", "/discovery/outages"

**Issue link:** 

- https://github.com/wso2-enterprise/wso2-ob-internal/issues/855

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** OB 3.0.0, CDS Toolkit

------

### Development Checklist

1. [ ] Built complete solution with pull request in place.
2. [ ] Ran checkstyle plugin with pull request in place.
3. [ ] Ran Findbugs plugin with pull request in place.
4. [ ] Ran FindSecurityBugs plugin and verified report.
5. [ ] Formatted code according to WSO2 code style.
6. [ ] Have you verify the PR does't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [ ] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [ ] Written unit tests.
2. [ ] Documented test scenarios(link available in guides).
3. [ ] Written automation tests (link available in guides).
4. [ ] Verified tests in multiple database environments (if applicable).
5. [ ] Verified tests in multiple deployed specifications (if applicable).
6. [ ] Tested with OBBI enabled  (if applicable).
7. [ ] Tested with specification regulatory conformance suites  (if applicable).

## Resources

**Knowledge Base:** https://sites.google.com/wso2.com/open-banking/

**Guides:** https://sites.google.com/wso2.com/open-banking/developer-guides

### CODE REVIEW

File: toolkits/ob-apim/repository/resources/wso2am-4.2.0-deployment-cds.toml [hunk 1]
```diff
@@ -532,6 +532,33 @@ priority = 8
 name = "org.wso2.openbanking.cds.gateway.executors.error.handler.CDSErrorHandler"
 priority = 1000
 
+[[open_banking.gateway.openbanking_gateway_executors.type]]
```
Comment (imesh94) [inline toolkits/ob-apim/repository/resources/wso2am-4.2.0-deployment-cds.toml:535]: Lets remove the unnecessary executors.

**Comment by imesh94** [comment]
Shall we change the logic of detecting Common endpoints along with this?

File: toolkits/ob-apim/repository/resources/wso2am-4.2.0-deployment-cds.toml (hunk 1 above)
Comment (aka4rKO) [reply toolkits/ob-apim/repository/resources/wso2am-4.2.0-deployment-cds.toml:535]: fixed with c334acc6a04c93c1baf13d40561200c8b105bfbf

**Comment by aka4rKO** [comment]
> Shall we change the logic of detecting Common endpoints along with this?

@imesh94 done with c334acc6a04c93c1baf13d40561200c8b105bfbf

**Review by imesh94 — APPROVED** [review]

**Review by imesh94 — APPROVED** [review]

**Review by anjuchamantha — APPROVED** [review]

**Pull request merged**